### ISSUE TITLE
Checking limits

### ISSUE THREAD

**Issue by iBreadInside** [s-yadav/react-number-format#598]
Hello!
Maybe somebody can help me. I am trying to make input with limits. Value need to be changed to max value if it excess.
I am trying to make it like this https://codesandbox.io/s/react-number-format-limits-qpk96?file=/src/App.js

But it refresh to max value only first time. If I press more numbers input value changes, but state is not.

**Comment by s-yadav** [comment]
Looks like a bug, Also, use isAllowed prop from limits, instead. Changing the number is Bad UX. better to prevent user to type if it doesn't validate.
https://codesandbox.io/s/react-number-format-limits-forked-8r4b8?file=/src/App.js

**Comment by iBreadInside** [comment]
> Looks like a bug, Also, use isAllowed prop for limits, instead. Changing the number is Bad UX. better to prevent user to type if it doesn't validate. https://codesandbox.io/s/react-number-format-limits-forked-8r4b8?file=/src/App.js

I'm not sure about it.
Let's look at that from the point of user who doesn't know how it works.
defaultValue = 150, max = 20000. User need to reach max.

1) Your method with isAllowed.
I am pushing the numbers and get smthg like 15024. But after that isAllow starts to block typing. Why? It is bugged? Am i reach max number of symbols in input? Even if I know max value - 20000, I don't understand, why I can't type greater value then it doesn't reach max. Only way for user is directly change exact numbers to make it possible.

2) My method. Let's imagine it is not bugged.
User just typing few more numbers and input automatically changes to max value. Profit.
Like, default 150 (push) 1503 (than) 15035 (and boom) 20000.

Maybe this is just my opinion, but I think it is less dificult and more user-friendly.

**Comment by s-yadav** [comment]
Yes, changing with blur is a better case. Also, a message for min and max values should be displayed to the user, so they know it beforehand.

Anyways, the library shouldn't limit the behaviour, so will definitely be fixing it.
